console/src/main/java/org/vaadin7/console/client/ConsoleConnector.java

```diff
@@ -204,6 +204,9 @@ public void onStateChanged(StateChangeEvent stateChangeEvent) {
             widget.getConfig().setGreeting(getState().greeting);
         if (!getState().history.isEmpty())
             widget.addPreviousHistory(getState().history);
+        if (!getState().shouldTrim != widget.getConfig().shouldTrim()){
```

Comment:
this is incorrect.  should be if (getState().shouldTrim != widget.getConfig().shouldTrim()) - remove the first "!"